### ISSUE TITLE
Unsupported Curve Type IfcIndexedPolyCurve

### ISSUE THREAD

**Issue by qwertybba** [xBimTeam/XbimGeometry#180]
when I try to convert IFC file generated from Tekla Structures it open without any problem.
But when I try to convert IFC4 file generated from Tekla Structures I get this error
Unsupported Curve Type IfcIndexedPolyCurve
 i use this code to convert IFC files to xBIM
` private void simpleButton1_Click(object sender, EventArgs e)
        {
            
            XtraOpenFileDialog ofd = new XtraOpenFileDialog();
            ofd.Filter = Resources.ifcName + "|*.ifc;*.ifczip;*.ifcxml|Xbim Files|*.xbim";
            ofd.Title = Resources.import3D;
            if (ofd.ShowDialog() == DialogResult.OK)
            {
                var xbimDbFilename = Path.ChangeExtension(ofd.FileName, "xBIM");
                // Make sure we are using an IModel implementation that supports saving of '.xbim' files
                IfcStore.ModelProviderFactory.UseHeuristicModelProvider();
                using (var model = IfcStore.Open(ofd.FileName))
                {
                    // IFC file is already parsed and open. Now build the 3D
                    var context = new Xbim3DModelContext(model);
                    context.CreateContext();    // Creates the Geometry using native GeometryEngine
                    // Save IFC to the internal XBIM format, which includes geometry
                    model.SaveAs(xbimDbFilename, StorageType.Xbim);
                }
            }

        }`
Whey I get this error and how can I get rid of it
Thanks in advance

**Comment by andyward** [comment]
Moved this from WindowsUI as its not related to the presentation

@qwertybba Can you supply the model with this issue? IfcIndexedPolyCurves came in with IFC4 Addendum 1 which should be implemented, but it's possible we're not handling some scenarios.

Also it would be good to confirm the version of Xbim.Geometry you're using.

@SteveLockley Should `IfcIndexedPolyCurve` (& IfcCompositeCurve) be handled in `XbimCurve::Init`?
https://github.com/xBimTeam/XbimGeometry/blob/master/Xbim.Geometry.Engine/XbimCurve.cpp#L141

**Comment by qwertybba** [comment]
> Moved this from WindowsUI as its not related to the presentation
> 
> @qwertybba Can you supply the model with this issue? IfcIndexedPolyCurves came in with IFC4 Addendum 1 which should be implemented, but it's possible we're not handling some scenarios.
> 
> Also it would be good to confirm the version of Xbim.Geometry you're using.
> 
> @SteveLockley Should `IfcIndexedPolyCurve` (& IfcCompositeCurve) be handled in `XbimCurve::Init`?
> https://github.com/xBimTeam/XbimGeometry/blob/master/Xbim.Geometry.Engine/XbimCurve.cpp#L141

this the link of the model
[https://drive.google.com/file/d/1QbOziufo6Z7Ub1wc73jJN2mQwkcwjJek/view?usp=sharing](url)
I use Tekla Structures 2019
Xbim.Geometry V5.0.163
xBIM Essentials V5.0.213

**Comment by andyward** [comment]
Confirmed issue in latest Geometry (216-develop). @SteveLockley  can you take a quick look while you're in the Geometry engine? I suspect it's just missing the handling of the 4x1 entitytype in that switch block.

```
   at Xbim.Geometry.XbimCurve2D.Init(IIfcCurve curve, ILogger logger)
   at Xbim.Geometry.XbimCurve2D..ctor(IIfcCurve curve, ILogger logger)
   at Xbim.Geometry.XbimGeometryCreator.CreateGrid(IIfcGrid grid, ILogger logger)
   at Xbim.Geometry.Engine.Interop.XbimGeometryEngine.CreateGrid(IIfcGrid grid, ILogger logger)
   at Xbim.ModelGeometry.Scene.Xbim3DModelContext.WriteShapeGeometries(XbimCreateContextHelper contextHelper, ReportProgressDelegate progDelegate, IGeometryStoreInitialiser geometryStore, XbimGeometryType geomStorageType)
   at Xbim.ModelGeometry.Scene.Xbim3DModelContext.CreateContext(ReportProgressDelegate progDelegate, Boolean adjustWcs)
   at XbimXplorer.XplorerMainWindow.OpenAcceptableExtension(Object s, DoWorkEventArgs args)
```

**Comment by SteveLockley** [comment]
This is fixed in the develop branch 5.0.219

**Comment by andyward** [comment]
Confirmed fixed
